### ISSUE TITLE
`Style/HashSlice` cop wrongly recommends slicing on a `.select` block argument

### ISSUE THREAD

**Issue by myronmarston** [rubocop/rubocop#13885]
## Expected behavior

I don't expect RuboCop to recommend `.slice` with an expression like this:

```ruby
results = referenced_types_by_source_type
  .select { |source_type, referenced_types| referenced_types.include?(source_type) }
```

The `.include?` inside the `.select` is using a block argument from the `.select` block itself. I don't believe there's any way to replace this with `.slice`, and I don't expect RuboCop to recommend it.

## Actual behavior

RuboCop recommends something that doesn't work (and can never work):

```
Inspecting 1 file
Scanning /Users/myron/Development/rubocop_bug.rb
For /Users/myron/Development: Default configuration from /Users/myron/.rvm/gems/ruby-3.3.4/gems/rubocop-1.72.2/config/default.yml
C

Offenses:

rubocop_bug.rb:16:4: C: [Correctable] Style/HashSlice: Use slice(*referenced_types) instead.
  .select { |source_type, referenced_types| referenced_types.include?(source_type) }
   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

...but `slice(*referenced_types)` can't work since `referenced_types` no longer exists when the `.select` block is removed.

## Steps to reproduce the problem

Put this into a Ruby script and run it.

```ruby
require "bundler/inline"

gemfile do
  source "https://rubygems.org"

  gem "rubocop", "1.72.2"
end

referenced_types_by_source_type = {
  "Foo" => ["Foo", "Bar"],
  "Bar" => [],
  "Bazz" => ["Bar"]
}

results = referenced_types_by_source_type
  .select { |source_type, referenced_types| referenced_types.include?(source_type) }

if ENV["PRINT"]
  puts results
end

require "rubocop"
RuboCop::CLI.new.run(["--format", "progress", "--only", "Style/HashSlice", "--debug", __FILE__])
```

## RuboCop version

1.72.2.
